packages/js/product-editor/src/components/block-editor/block-editor.tsx

```diff
@@ -119,18 +121,20 @@ export function BlockEditor( {
 		};
 	}, [ canUserCreateMedia, _settings ] );
 
-	const [ productType ] = useProductEntityProp< Product[ 'type' ] >( 'type', {
-		postType,
-	} );
-
 	const [ productTemplateId ] = useProductEntityProp< string >(
 		'meta_data._product_template_id',
 		{ postType }
 	);
 
+	const { record: product } = useEntityRecord(
+		'postType',
+		postType,
+		productId
+	);
```

Comment:
We might want to handle `isResolving` here to pass appropriate value of product into `useProductTemplate`